apps/meteor/server/services/team/service.ts

```diff
@@ -137,7 +138,14 @@ export class TeamService extends ServiceClassInternal implements ITeamService {
 				_id: teamId,
 				...teamData,
 			};
-		} catch (e) {
+		} catch (error) {
+			// Manually deleting team if an error occurs, this is a temporary fix due to urgency
+			// to release version 6.12 and will be refactored on next release.
+			if (teamId) {
+				await Team.deleteOneById(teamId);
+				await TeamMember.deleteByTeamId(teamId);
```

Comment:
If this happens while converting a room into a team, then the room will still retain the invalid `teamId` set by `Rooms.setTeamMainById`. Do you think that's possible and we should take care of? Or nah?

Reply:
The team should be valid in this case.

But it's definetly worth adding a way to handle this... I'm open to suggestions, tho. haha

Reply:
The team should be valid in this case.

But it's definetly worth adding a way to handle this... I'm open to suggestions, tho. haha

Reply:
IMO it feels ok not to cover this since I believe we're only trying to cover errors "thrown by RC" here, not by the DB itself (in this scenario, `Room.create` is the only part that is really prone to errors here). If we'd like to cover errors from the DB here, then there'd be much more to undo (such as deleting messages, unsetting room as team's main, deciding if we should delete the room...). wdyt @KevLehman ?
Please correct me if I'm worng, @lucas-a-pelegrino

Reply:
That's correct @matheusbsilva137 

This code I've added aims to avoid teams being created with no rooms associated with it for release 6.12.

If a room is being converted to a team, it's already created, therefore valid.

This catch block where I delete the Team and TeamMembers from the db will be removed once I've designed the final solution in the next release. We will have a robust solution capable of handling errors appropriately. That will involve more time to refactor that we don't have now, though.

Reply:
all right, thanks guys!